portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -0,0 +1,12 @@
+package com.google.sps.data;
+
+public class Comment {
+  public String content;
```

Comment:
I understand that this class is not an AutoValue class because you need to parse a json string using `gson.fromJson(body, Comment.class);`. Technically you can use `gson.fromJson(body, AutoValue_Comment.class);` but I'm not sure if this is discouraged. @fenghaolw Hao, do you know? 

However, maybe avoid making these fields public?  One suggestion is to set these fields to private, and only create a getter for them, effectively making these fields "readonly". I agree that creating these boiler plate code is tedious, which is why AutoValue is helpful :)

Reply:
I have never used Gson and AutoValue together before. I think what you described should work, referencing a generated class name is a little weird, I don't know it matters though.

A quick search also shows that people were trying to do the same thing: https://github.com/rharter/auto-value-gson :D It is also imported in google3 and be used in a fair amount of internal codebase, so you could also try that.

Reply:
Just to confirm, is it just
`  public static TypeAdapter<Comment> typeAdapter(Gson gson) {
    return new AutoValue_Comment.GsonTypeAdapter(gson);
  }`
and then you could just call `Comment.TypeAdapter(gson)` (returning a TypeAdapter<Comment> rather than a Comment?)

Reply:
also yes! made the change about private fields haha honestly I don't know why I even made them public in the first place my bad!

Reply:
Would it be better to use AutoValue_Comment.class? (Is there a reason that this is possibly discouraged?)

Reply:
I think (from just reading the readme) that after you add this static typeAdapter method to you Comment AutoValue class, it allows you to `gson.fromJson(body, Comment.class);` whereas a normal AutoValue class does not. However from the readme I didn't see how to import it with Maven...

Using `AutoValue_Comment.class` is weird but I guess not explicitly discouraged.

Reply:
should I do that instead of what I'm currently using?

Reply:
Up to you!